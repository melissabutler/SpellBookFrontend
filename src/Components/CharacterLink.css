.CharacterLink {
    color: black;
    background-color: white;
    margin: 20px;
    padding: 10px;
    border-radius: 10px;
    
    box-shadow: 3px 3px 3px black;

}

h3 {
    padding: 10px;
    margin: 10px;
}
