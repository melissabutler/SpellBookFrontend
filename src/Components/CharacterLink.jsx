import React from "react";

import "./CharacterLink.css"

import { Container, Row, Col } from 'react-bootstrap';

const CharacterLink = ({character}) => {

    let charClass = character.char_class[0].toUpperCase() + character.char_class.slice(1);


    return (
            <Container className="CharacterLink">
                <Row>
                    <Col>
                        <h3 className="CharacterLink-name">{character.char_name}</h3>
                    </Col>
                    <Col>
                        <h3>Level {character.lvl} {charClass}</h3>
                    </Col>
                    
                </Row>
                
            </Container>
    )
}

export default CharacterLink;