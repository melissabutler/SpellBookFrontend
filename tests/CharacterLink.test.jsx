import React from "react";
import { describe, it, expect, vi } from "vitest";
import { render, screen} from '@testing-library/react'
import CharacterLink from "../src/Components/CharacterLink";

describe("CharacterLink", function() {
        vi.mock('react', async (importOriginal) => {
            const react = await importOriginal('react');
            return {
                ...react,
                useContext: () => ({ currentUser: {
                    username: 'u1',
                    email: 'email@email.com',
                    isAdmin: false,
                    characters: [
                        {char_class: "cleric", char_name: "Beebo",
                        id: 1,
                        lvl: 3
                    },
                    {
                        char_class: "cleric",
                        char_name: "Beebo",
                        id: 2,
                        lvl: 3
                    }
                    ]
                }})
        }})

        vi.mock('react-router-dom', async () => {
            const mod = await vi.importActual('react-router-dom');
            return {
                ...mod, 
                useParams: () => ({
                    id: 1,
                }),
                useNavigate: () => "/"
            }
        })
    it('renders the CharacterLink component', async () => {
        render(<CharacterLink character={ {char_class: "cleric", char_name: "Beebo", id: 2, lvl: 3}}/>)
        screen.debug();
      
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<CharacterLink character={ {char_class: "cleric", char_name: "Beebo", id: 2, lvl: 3}}/> );
        expect(asFragment()).toMatchSnapshot();
    })

    it("displays input info", async () => {
        render(<CharacterLink character={ {char_class: "cleric", char_name: "Beebo", id: 2, lvl: 3}}/>)

        expect(screen.getByText("Level 3 Cleric")).toBeInTheDocument();
        expect(screen.getByText("Beebo")).toBeInTheDocument();
    })


})
   