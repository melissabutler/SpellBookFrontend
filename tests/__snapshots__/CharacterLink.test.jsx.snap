// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CharacterLink > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CharacterLink container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3
          class="CharacterLink-name"
        >
          Beebo
        </h3>
      </div>
      <div
        class="col"
      >
        <h3>
          Level 3 Cleric
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
